Update documentations of Configuration default environments (#3321)
kedro-org/kedro: docs/source/configuration/configuration_basics.md

## Configuration environments
A configuration environment is a way of organising your configuration settings for different stages of your data pipeline. For example, you might have different settings for development, testing, and production environments.

> By default, Kedro Framework has a `base` and a `local` environment.

Author: I didn't change the wordings too much. In most of the configuration docs, we still saying `OmegaConfigLoader` will load config in `base` and `local`, technically this is only true for Kedro Framework users. I think this will be the majority.

Instead, I add a bit more explanation in the `advance_configuration.md` to explains the different defaults of environment. Feel free to comment on this.

Reviewer: I'd maybe go with the following:
```suggestion
By default, Kedro projects have a `base` and a `local` environment.
```
